src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -645,10 +645,10 @@ private TNode AddFailureElement(TNode targetNode)
         {
             TNode failureNode = targetNode.AddElement("failure");
 
-            if (Message != null)
+            if (Message != null && Message.Trim().Length > 0)
                 failureNode.AddElementWithCDATA("message", Message);
 
-            if (StackTrace != null)
+            if (StackTrace != null && StackTrace.Trim().Length > 0)
                 failureNode.AddElementWithCDATA("stack-trace", StackTrace);
 
             return failureNode;
```

Comment:
Yeah - one way to get it would be to simply call Assert.Fail(), Assert.Fail(null), Assert.Fail(string.Empty), etc. Shouldn't this still result a test failure element, even if empty?